### DISCUSSION TITLE
How to use scripting feature in page

### DISCUSSION THREAD

**Discussion by syedshadab19** [OrchardCMS/OrchardCore#11935] [Q&A]
Hi i see that we have scripting feature as shown below in snapshot. I want to know how to use it on a page so that I can add javascript to  the page.

![image](https://user-images.githubusercontent.com/25971751/176881628-5ff6ce0c-c72f-441b-b420-055b1ce3fcf6.png)

**Comment by hishamco** [comment]
Have a look to the docs https://docs.orchardcore.net/en/latest/docs/reference/modules/Scripting/

**Reply by syedshadab19** [reply]
Is there a way to use scripting directly on a page using UI?

**Reply by hishamco** [reply]
Could you please elaborate?

**Reply by syedshadab19** [reply]
Fox example i wanted to know if there is a way to add javascript using user interface like how we can build a page by adding HTML  to the HTML body text box as shown in the below snaphot. Is there a similar method for adding javascript also like how we are adding the HTML body.

![image](https://user-images.githubusercontent.com/25971751/177129983-ac54abcd-fca6-4d34-a151-8d4c510c8712.png)

**Reply by hishamco** [reply]
Javscript code is blocked for security purpose, I'm not sure why you need the users to provide them?

**Reply by syedshadab19** [reply]
ok to add some additional functionalities like adding popup on a page

**Reply by hishamco** [reply]
We already used popups in many places, at the end we don't advice to enable scripts from user input, but you can disable santizer feature, so everything will be under your risk

**Reply by syedshadab19** [reply]
ok can you please share the file name where the sanitizer is present in orchadcore?

**Reply by hishamco** [reply]
https://github.com/OrchardCMS/OrchardCore/blob/339d74b251b53393dd510f8ad3027aa17d78500d/src/OrchardCore/OrchardCore.Infrastructure/Html/HtmlSanitizerService.cs

**Comment by ns8482e** [comment]
The scripting feature allows you to run JavaScript on server.  You can utilize this feature and can add some script logic to workflow, recipe or in your c# code.

This scripting feature is not related to html client side script.  Client side script/js files can be added in to `wwwroot’ folder and reference using script tag in html/liquid

As @hishamco said, OC OOTB doesn’t support  adding script tag from admin inside html/liquid editor due to potential security risk. However you can configure sanitizer to allow script tag in startup, which not recommended

**Reply by Skrypt** [reply]
Liquid supports script tags.

```liquid
{% scriptblock at: "Foot" %}
    document.write('<!-- some script -->');
{% endscriptblock %}
```

But it needs to be processed on server-side through the usage of a Liquid tag.

https://docs.orchardcore.net/en/latest/docs/reference/modules/Resources/#custom-scripts

**Reply by syedshadab19** [reply]
Thanks for the response
